packages/fxa-auth-server/test/oauth/jwt_access_token.js

```diff
@@ -118,6 +118,63 @@ describe('lib/jwt_access_token', () => {
 
       assert.equal(signedClaims.aud, TOKEN_SERVER_URL);
     });
+
+    describe('PPID / uid scope tests', () => {
+      beforeEach(() => {
+        requestedGrant.ppidSeed = 1000;
+      });
+
+      const scopeTest = async (scopes) => {
+        requestedGrant.scope = ScopeSet.fromArray(scopes);
+        try {
+          await JWTAccessToken.create(mockAccessToken, requestedGrant);
+          assert.fail();
+        } catch (err) {
+          assert.instanceOf(err, AppError);
+          assert.equal(err.errno, 123);
+        }
+      };
+
+      it('should not allow a PPID token to grant access to a profile:uid scope', async () => {
+        await scopeTest(['profile:uid']);
+      });
+
+      it('should not allow a PPID token to grant access to a profile scope', async () => {
+        await scopeTest(['profile']);
+      });
+
+      it('should not allow a PPID token to grant access to a profile:write scope', async () => {
+        await scopeTest(['profile:write']);
+      });
+
+      it('should not allow a PPID token to grant access to a profile:write sub-scope', async () => {
+        await scopeTest(['profile:write:avatar']);
```

Comment:
You want `profile:avatar:write` here; my word I dislike how we did the `write` suffix on these initial set of scopes...

Reply:
doh, thanks!